proxy/src/control/pb.rs

```diff
@@ -66,13 +68,15 @@ impl event::StreamResponseFail {
     fn to_tap_event(&self, ctx: &Arc<ctx::http::Request>) -> common::TapEvent {
         use self::common::tap_event;
 
+        let since_request_open = self.response_fail_at - self.request_open_at;
+        let since_response_open = self.response_fail_at - self.response_open_at;
         let end = tap_event::http::ResponseEnd {
             id: Some(tap_event::http::StreamId {
                 base: 0, // TODO FIXME
                 stream: ctx.id as u64,
             }),
-            since_request_init: Some(pb_duration(&self.since_request_open)),
-            since_response_init: Some(pb_duration(&self.since_response_open)),
+            since_request_init: Some(pb_duration(&since_request_open)),
```

Comment:
Maybe `pb_duration()` should just take two instants and do the math itself?

Reply:
I've added a `pb_elapsed` variant that does this